From 2146bb728b91f0d0e530061555ab704ed98bf8c6 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 04 Apr 2022 13:04:08 +0300
Subject: [PATCH] [SW-79580] greco: change preboot size to 63K

This patch changes preboot size to 63K. Currently PREBOOT_MAX_SIZE
is already set to 63K, so changing macro PREBOOT_SIZE to 63K also,
to be aligned.
Also due to hw errata  spi dma copy size should not be greater
that 63K, more then that, it should be in multiple copies.
Since current greco preboot size(approc 50K) is well under 63K,
multi-copy support is not required as of now.

Change-Id: I71ef6e4d5d2c1a8fb0f72e8b38e0fdbe05e6f9ae
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index f83b41e..08b0a3a 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -10,7 +10,7 @@
 #include <sizes.h>
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
-#define PREBOOT_SIZE			0x10400		/* 65KB */
+#define PREBOOT_SIZE			0xFC00		/* 63KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index ada859f..28af0b4 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -12,7 +12,7 @@
 #include <sizes.h>
 
 #define PPBOOT_SIZE		0x4000		/* 16KB */
-#define PREBOOT_SIZE		0x10400		/* 65KB */
+#define PREBOOT_SIZE		0xFC00		/* 63KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
